packages/web-components/fast-foundation/src/date-picker/date-picker.template.ts

```diff
@@ -0,0 +1,324 @@
+import { Calendar } from "../calendar";
+import { DataGrid, DataGridCell, DataGridRow } from "../data-grid";
+import { ListboxElement } from "../listbox/";
+import { ListboxOption } from "../listbox-option";
+import { TextField } from "../text-field";
+import type { DatePicker, DatePickerOptions } from "./date-picker";
+
+/**
+ * Template to render a time picker menu
+ * @param context - Control context
+ * @param times - labels and values for times, hours, minutes and meridian
+ * @returns - A time picker template
+ * @public
+ */
+export const timePickerTemplate = (
+    context: ElementDefinitionContext,
+    times,
+    timeKeydown
+) => {
+    const listbox = context.tagFor(ListboxElement);
+    const listboxOption = context.tagFor(ListboxOption);
+    return html`
+        <div class="time-picker">
+            <${listbox}
+                class="time-list"
+                ${ref("hourSelect")}
+                size="7"
+                @keydown="${(x, c) => timeKeydown("hour", c.event)}"
+            >
+                ${repeat(
+                    () => times.hours,
+                    html`
+                        <${listboxOption} @click="${x => x.action()}">
+                            ${x => x.text}
+                        </${listboxOption}>
+                    `
+                )}
+            </${listbox}>
+            <div class="time-separate">:</div>
+            <${listbox}
+                class="time-list"
+                ${ref("minuteSelect")}
+                size="7"
+                @keydown="${(x, c) => timeKeydown("minute", c.event)}"
+            >
+                ${repeat(
+                    () => times.minutes,
+                    html`
+                        <${listboxOption} @click="${x => x.action()}">
+                            ${x => x.text}
+                        </${listboxOption}>
+                `
+                )}
+            </${listbox}>
+            <${listbox}
+                class="time-list"
+                ${ref("meridianSelect")}
+                size="7"
+                @keydown="${(x, c) => timeKeydown("meridian", c.event)}"
+            >
+                ${repeat(
+                    () => times.meridians,
+                    html`
+                        <${listboxOption} @click="${x => x.action()}">
+                            ${x => x.text}
+                        </${listboxOption}>
+                `
+                )}
+            </${listbox}>
+        </div>
+    `;
+};
+
+/**
+ *
+ * @param context - The date picker context
+ * @param items  - An array of items to display
+ * @param title - The picker title text
+ * @param previousAction - Action to show previous options
+ * @param nextAction - Action to show next options
+ * @param reset - Reset views action
+ * @returns - A picker template used by month and year pickers
+ * @public
+ */
+const pickerTemplate = (context, items, title, previousAction, nextAction, reset) => {
```

Comment:
I figured out how this comes together, but I wonder if that could be the responsibility of the calendar. This is complicated because we allow slotting the calendar, which I get from a principle perspective but not sure practically what someone would be doing there. My concern is around being able to have a calendar that's much different than a month or year picker. In most cases those would not be side by side, so probably doesn't matter much, but poses some issues for styling consistency. Also, it feels like our base calendar should have those month and year modes for general calendar navigation like other example components do, allowing to drill up to month then year for instance.

Reply:
The calendar is no longer slotted. I could make a breaking out the month and year pickers out into their own component but that should probably be a separate work item.